### PR TITLE
Drop support php 7, add php 8.2 support

### PR DESCRIPTION
**Pull request by fezfez**

### CODE REVIEW

File: src/AbstractConfigFactory.php [hunk 1]
```diff
@@ -52,11 +43,10 @@ public function canCreateServiceWithName(ServiceLocatorInterface $serviceLocator
     /**
      * Determine if we can create a service (SM v3)
      *
-     * @param ContainerInterface $container
      * @param string $requestedName
      * @return bool
      */
-    public function canCreate(ContainerInterface $container, $requestedName)
+    public function canCreate(containerinterface $container, $requestedName)
```
Comment (Ocramius) [inline src/AbstractConfigFactory.php]: This case sensitivity change is a bug in PHPCS, I think

File: src/StandaloneReaderPluginManager.php [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 class StandaloneReaderPluginManager implements ContainerInterface
 {
+    /** @var string[] */
```
Comment (Ocramius) [inline src/StandaloneReaderPluginManager.php]: We should be a bit more precise about the keys here

File: src/StandaloneWriterPluginManager.php [hunk 3]
```diff
@@ -13,6 +13,7 @@
 
 class StandaloneWriterPluginManager implements ContainerInterface
 {
+    /** @var string[] */
```
Comment (Ocramius) [inline src/StandaloneWriterPluginManager.php]: We should be a bit more precise about the keys here

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/AbstractConfigFactory.php (hunk 1 above)
Comment (fezfez) [reply src/AbstractConfigFactory.php]: yep

```
FILE: src/AbstractConfigFactory.php
----------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------------------
 5 | ERROR | [x] Expected class name interop\container\containerinterface; found Interop\Container\Containerinterface
----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------------------------------

```

Comment (fezfez) [reply src/AbstractConfigFactory.php]: i ignore this in phpcs.xml

File: src/StandaloneReaderPluginManager.php (hunk 2 above)
Comment (fezfez) [reply src/StandaloneReaderPluginManager.php]: done

File: src/StandaloneWriterPluginManager.php (hunk 3 above)
Comment (fezfez) [reply src/StandaloneWriterPluginManager.php]: done

**Comment by fezfez** [comment]
@Ocramius should be ok

**Review by Ocramius — APPROVED** [review]
Thanks @fezfez!

**Pull request merged**